### DISCUSSION TITLE
issues with gcc

### DISCUSSION THREAD

**Discussion by ipatch** [Homebrew/discussions#1961] [Everyday usage] [answered]
I'm working with a _big sur_ box, and am attempting to install freecad from the tip of the master branch (latest commit). I have successfully installed all dependencies for freecad using

```
brew install freecad/freecad/freecad --only-dependencies
```

the above cmd goes through and installs all the deps required to build freecad from git source. i then proceed to do the,

```
cmake [OMIT_ADDITIAL_ARGS_FOR_BREVITY]
make;
make install;
```

however during the `make install` process, when assembling the `FreeCAD.app` apple bundle, the _install_ process is searching for a library provided by the gcc homebrew formula/package (presuming).

and i have noticed that the latest `gcc` formula installs and links more files than the `gcc@10` formula.

i have both gcc versions installed, and i have linked both versions durning the `cmake; make; make install` process. however during the `make install` process it's looking for the below file which does not exist on my system

```
/usr/local/opt/gcc/lib/gcc/10/libquadmath.0.dylib
```

which will return a, `ERROR: Failed to resolve dependency`

the above file however does exist in the `gcc` packages, but in the `gcc@10` package, the path differs from where the install is searching for the lib.

within the current gcc package ie. gcc11 there is a `libquadmath.0.dylib` but not within the `gcc@10` package.

doing a quick search for `libquadmath` within `/usr/local` returns

```
Cellar/gcc/11.2.0/lib/gcc/11/libquadmath.a
Cellar/gcc/11.2.0/lib/gcc/11/libquadmath.0.dylib
Cellar/gcc@10/10.3.0/lib/gcc/10/libquadmath.0.dylib
Cellar/gcc@10/10.3.0/lib/gcc/10/libquadmath.a
```

_is this the intended behavior or am i doing something wrong here?_

**Comment by carlocab** [comment, marked as answer]
This looks like an issue with the freecad build system, no?

If it needs `libquadmath.0.dylib`, and is using `gcc@10`, then it needs to be looking either in `/usr/local/lib/gcc/10` or `/usr/local/opt/gcc@10/lib/gcc/10`. If it's using `gcc`, then it needs to be looking in either of `/usr/local/lib/gcc/11`, or `/usr/local/opt/gcc/lib/gcc/11`.

It's very unlikely that the path `/usr/local/opt/gcc/lib/gcc/10` came from either `gcc` or `gcc@10`, so it seems to me that the problem is in freecad.
